azure_pipeline.yaml

```
@@ -13,7 +13,64 @@ variables:
   action:
 
 stages:
+  - stage: Precheck
+    jobs:
+      - job: Validate
+        pool:
+          vmImage: $(agentPool)
+        steps:
+        - task: AzureKeyVault@1
+          displayName: Retrieve keyvault secret for ADO token
+          inputs:
+            ConnectedServiceName: azurerm-sandbox
+            keyVaultName: infra-vault-nonprod
+            secretsFilter: 'azure-devops-token'
+            runAsPreJob: false
+        - task: Bash@3
+          displayName: Prevent parallel run
+          env:
+            thisbuild: $(Build.BuildId)
+            pipelinedefinition: $(System.DefinitionId)
+          inputs:
+            targetType: 'inline'
+            script: | 
+              #!/bin/bash
+              set -e
+              organization="hmcts"
+              project="CNP"
+
+              echo "This is build $thisbuild"
+              IFS=$'\n'
+              JSON_DATA=($(curl -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u | grep inProgress))
```

logs are getting spammed because of this
```suggestion
              JSON_DATA=($(curl -s -u :$(azure-devops-token) --request GET "https://dev.azure.com/$organization/$project/_apis/build/builds?api-version=5.1&definitions=$pipelinedefinition" -H "Content-Type: application/json" | jq  '.value[] | .status + (.id|tostring)' | sort -u | grep inProgress))
```